larq/quantizers.py

```diff
@@ -74,17 +74,13 @@
 ]
 
 
-@tf.custom_gradient
-def _binarize_with_identity_grad(x):
-    return math.sign(x), lambda dy: dy
-
-
-@tf.custom_gradient
-def _binarize_with_weighted_grad(x):
-    def grad(dy):
-        return (1 - tf.abs(x)) * 2 * dy
+def _clipped_gradient(x, dy, clip_value):
+    if clip_value is None:
+        return dy
 
-    return math.sign(x), grad
+    zeros = tf.zeros_like(dy)
+    mask = tf.math.less_equal(tf.math.abs(x), clip_value)
```

Comment:
This could also be done using
```python
mask = tf.math.logical_and(
        tf.math.less_equal(x, clip_value), tf.math.greater_equal(x, -clip_value)
    )
```
I roughly tested this on a GPU where usign `tf.abs` was faster. This solution is slightly faster on my non optimized mac CPU build.

Reply:
would keep the current implementation as it's a bit cleaner.